src/scenario.jl

```diff
@@ -668,12 +670,18 @@ function run_model(domain::Domain, param_set::YAXArray, functional_groups::Vecto
             site_data.area .* site_data.k, (1, 1, n_locs)
         )
 
+        # Constrain growth by available area
+        # Constraint Coefficient -> log(1 + left over space)
+        growth_spatial_constraint .= log.(2, 1 .+ relative_leftover_space(
+            dropdims(sum(C_t; dims=(1, 2)), dims=(1, 2))
+        ))
```

Comment:
Hmm... not sure how much this will change things but technically this could be moved into the loop over locations below.

Keep the `growth_spatial_constraint` but make it so each thread is calculating/updating/using the cache separately.

This way we're only looping over $n$ (where $n$ is the number of reefs) once instead of twice.